Create "TranslateSyncFlags.py"

geransmith opened the pull request:
This script translates the syncFlags bitmask into human readable text. Can be used as an example for querying data via the Bit9 API.

geransmith committed:
Create "TranslateSyncFlags.py"

This script translates the syncFlags bitmask into human readable text. Can be used as an example for querying data via the Bit9 API.

jgarman commented on bit9PlatformAPI/examples/samples/TranslateSyncFlags.py:
```diff
@@ -0,0 +1,30 @@
+)
+
+# Find all computers with the specified name that are connected
+comps = bit9.search('v1/computer', ['name:DOMAIN\NAME', 'connected:true'])
```
Can you generalize this into two parts:
1. A function that can be reused by others (eg. `def convert_sync_flags(sync_flags)` which takes the hex value from the computer's `syncFlags` attribute and returns a list of strings with the converted flag values)
2. Refactor into a simple script driver that takes a command line parameter describing the search parameters (computer name, etc.) that are passed to the `bit9.search()` function. Then that main function performs the search and passes the syncFlags into the function defined above, to show the proper usage of the function, and formats the output as you have here.

Thanks!

jgarman commented:
Thanks for the contribution! Address the comments above and I'll merge your pull request.

geransmith committed:
Update to match Jason's suggested changes

Should meet Jason's requirement here: https://github.com/carbonblack/bit9platform/pull/5

Can you generalize this into two parts:

A function that can be reused by others (eg. def convert_sync_flags(sync_flags) which takes the hex value from the computer's syncFlags attribute and returns a list of strings with the converted flag values)
Refactor into a simple script driver that takes a command line parameter describing the search parameters (computer name, etc.) that are passed to the bit9.search() function. Then that main function performs the search and passes the syncFlags into the function defined above, to show the proper usage of the function, and formats the output as you have here.
Thanks!

tehspaceg commented:
@jgarman, I made the changes you suggested. Let me know if you want any other changes done there. I think I covered everything.

Sorry if any of the code isn't "Python" enough, still learning things.
